tilelang/jit/adapter/nvrtc/libgen.py

```
@@ -0,0 +1,235 @@
+
+        # Ensure the context is valid
+        ctx = cuda.cuCtxGetCurrent()[1]
+        if cuda.cuCtxGetApiVersion(ctx)[0] != cuda.CUresult.CUDA_SUCCESS:
+            import torch
+            torch.cuda.synchronize()
+
+        result, self.culib = cuda.cuLibraryLoadFromFile(
+            bytes(lib_path, "utf-8"), [], [], 0, [], [], 0)
+        assert result == cuda.CUresult.CUDA_SUCCESS, f"Failed to load library: {lib_path}"
+
+    def compile_lib(self, timeout: float = None):
```

Ruff (0.14.4)

Unused method argument: `timeout`

(ARG002)